galaxyproject/galaxy
# Add onedata objectstore

Object store plugin for Onedata distributed data management system (https://onedata.org/). Requires Onedata version >= 21.02.5.

## How to test the changes?
(Select all options that apply)
    1. Add the `onedatafilerestclient` package to the currently used requirements (can be found in conditional-requirements.txt).
 
    2. Create `object_store_conf.xml` file in the config dir, copying the part concerning Onedata from `object_store_conf.xml.sample`.
    Fill in the config (`access_token`, `onezone_domain`, `space`, `disable_tls_certificate_validation` - you may want `True` - see below).

        You will need an account in a Onedata system instance, however, version 21.02.5 is not officially released yet. The best way to test is to use the "demo mode", which allows easily setting up a dockerized Onedata environment: https://onedata.org/#/home/documentation/topic/stable/demo-mode

    3. Create galaxy.yml file in the config dir with the following content:

    ```
    galaxy:
      object_store_config_file: object_store_conf.xml
    ```

    4. Start the galaxy server, log in.

    5. Upload some data and run some jobs. Open the Onedata UI and navigate to your space to see the stored Galaxy datasets.

## License
